examples/m2m-auth-with-next/app/actions.ts

```diff
@@ -0,0 +1,59 @@
+"use server";
+
+import { kv } from "@vercel/kv";
+import { revalidatePath, unstable_noStore } from "next/cache";
+import { TimeSpan } from "oslo";
+import { HMAC, alphabet, generateRandomString } from "oslo/crypto";
+import { createJWT } from "oslo/jwt";
+
+interface ApiKey {
+  keyId: string;
+  expiresIn: TimeSpan;
+}
+
+/**
+ * For the purposes of this example, API keys are going to be stored and retrieved from Vercel KV
+ */
+export async function fetchActiveApiKeys() {
+  // Recently created keys should be reflected in the UI, therefore opting out for not caching it
+  unstable_noStore();
+
+  const keys = await kv.smembers<Array<ApiKey>>("api-keys");
+
+  return keys;
+}
+
+export async function createApiKey() {
```

Comment:
Will rollback all of this tomorrow. Instead of manually creating JWTs and storing into Vercel KV, will take a simpler approach to use `unkey` behind the scenes and abstract in a way that Clerk would be handling this.